### PR TITLE
Explorer TX pagination

### PR DESCRIPTION
**Pull request by Jibz1**
Page pagination added. 

- Switch from `getRecentTransactions` to `getTransactionDigestsInRange` on RecentTx can 
- Refactored/ combined `getRecentTransactions` so that it is called once on the `Home.tsx` and the data shared across component
- Added URL params to paginate `?p=1` 
 

https://user-images.githubusercontent.com/8676844/171495077-264566b1-3974-4027-b51d-f612aec27d56.mov

### CODE REVIEW

**Comment by Jibz1** [comment]
@bmwill `getTotalTransactionNumber` is called when the page load.  The pagination uses the sequence number to request transactions within the sequence range. So even if a new stream of transactions flows in, the total number of transactions may increase but the sequence numbers should be the same.

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 1]
```diff
@@ -218,17 +255,27 @@ function LatestTxCardStatic() {
...
     const [network] = useContext(NetworkContext);
+    const [searchParams] = useSearchParams();
+    const [txNumPerPage] = useState(15);
     useEffect(() => {
         let isMounted = true;
-        getRecentTransactions(network, 15)
-            .then((resp: any) => {
+        const pagedNum: number = parseInt(searchParams.get('p') || '1', 10);
+
+        // pass the network to the API call, totaltx count, number of transactions per page and page number
```
Comment (stella3d) [inline explorer/client/src/components/transaction-card/RecentTxCard.tsx:276]:
```suggestion
```

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 2]
```diff
@@ -218,17 +255,27 @@ function LatestTxCardStatic() {
         loadState: 'loaded',
         latestTx: latestTx,
     };
-    return <LatestTxView results={results} />;
+    return (
+        <>
+            <LatestTxView results={results} />
+            <Pagination totalTxCount={350} txNum={15} />
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:261]: @stella3d random total tx count number for e2e

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx [hunk 3]
```diff
@@ -49,14 +50,50 @@ type TxnData = {
...
     network: Network | string,
-    txNum: number
+    totalTx: number,
+    txNum: number,
+    pageNum?: number
 ): Promise<TxnData[]> {
     try {
         // Get the latest transactions
+
+        // TODO: Get total transactions once across conmponents
```
[the remark this answers is not in the document]
Comment (Jibz1) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:80]: I will remove that comment.

File: explorer/client/src/components/transaction-card/RecentTxCard.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (stella3d) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:260]: is it only used in the static e2e environment, or in the real site ? hard to tell from the diff

[the remark this answers is not in the document]
Comment (Jibz1) [reply explorer/client/src/components/transaction-card/RecentTxCard.tsx:260]: Screen Shot 2022-06-01 at 5 44 42 PM
Screen Shot 2022-06-01 at 5 48 08 PM

**Review by stella3d — APPROVED** [review]

**Pull request merged**